### PR TITLE
Revert to Kotlin 1.5.10

### PR DESCRIPTION
**Pull request by nhachicha**
Kotlin 1.5.20 causing some incompatibility issues with the Bookshelf app 
```
e: Could not find "/Users/nabil/.gradle/caches/modules-2/files-2.1/io.realm.kotlin/library-iosx64/0.45.0-SNAPSHOT/9f6174f4f610fd23472aa2f0e2d9879584c9ecb5/library.klib" in [/Users/nabil/Dev/realm/KMP/master-2/realm-kotlin-samples/Bookshelf, /Users/nabil/.konan/klib, /Users/nabil/.konan/kotlin-native-prebuilt-macos-1.5.10/klib/common, /Users/nabil/.konan/kotlin-native-prebuilt-macos-1.5.10/klib/platform/ios_x64]

The message received from the daemon indicates that the daemon has disappeared.
Build request sent: Build{id=9ff73205-f03d-4bcd-99a6-25f7b424d5f7, currentDir=/Users/nabil/Dev/realm/KMP/master-2/realm-kotlin-samples/Bookshelf}
Attempting to read last messages from the daemon log...
Daemon pid: 47119
  log file: /Users/nabil/.gradle/daemon/7.0.2/daemon-47119.out.log
----- Last  20 lines from daemon log file - daemon-47119.out.log -----
2021-07-13T14:16:00.759+0100 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] daemon is running. Sleeping until state changes.
2021-07-13T14:16:00.759+0100 [INFO] [org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy] Daemon is about to start building Build{id=9ff73205-f03d-4bcd-99a6-25f7b424d5f7, currentDir=/Users/nabil/Dev/realm/KMP/master-2/realm-kotlin-samples/Bookshelf}. Dispatching build started information...
2021-07-13T14:16:00.760+0100 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 29: dispatching org.gradle.launcher.daemon.protocol.BuildStarted@70073589
2021-07-13T14:16:00.760+0100 [DEBUG] [org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment] Configuring env variables: [PATH, JAVA_HOME, GITHUB_DOCKER_TOKEN, TERM, COMMAND_MODE, __INTELLIJ_COMMAND_HISTFILE__, ANDROID_HOME, NDK_HOME, GITHUB_DOCKER_USER, P9K_TTY, LOGNAME, APP_NAME_47126, XPC_SERVICE_NAME, PWD, __CFBundleIdentifier, SHELL, PAGER, LSCOLORS, ANDROID_NDK_HOME, OLDPWD, USER, ANDROID_SDK_ROOT, ANDROID_NDK, ZSH, JAVA_MAIN_CLASS_47126, P9K_SSH, TERMINAL_EMULATOR, LOGIN_SHELL, TMPDIR, SSH_AUTH_SOCK, XPC_FLAGS, __CF_USER_TEXT_ENCODING, LESS, LC_CTYPE, APP_ICON_47126, HOME, SHLVL]
2021-07-13T14:16:00.761+0100 [DEBUG] [org.gradle.launcher.daemon.server.exec.LogToClient] About to start relaying all logs to the client via the connection.
2021-07-13T14:16:00.761+0100 [INFO] [org.gradle.launcher.daemon.server.exec.LogToClient] The client will now receive all logging from the daemon (pid: 47119). The daemon log file: /Users/nabil/.gradle/daemon/7.0.2/daemon-47119.out.log
2021-07-13T14:16:00.762+0100 [INFO] [org.gradle.launcher.daemon.server.exec.LogAndCheckHealth] Starting 2nd build in daemon [uptime: 5.694 secs, performance: 100%]
2021-07-13T14:16:00.762+0100 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] The daemon has started executing the build.
2021-07-13T14:16:00.762+0100 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=2286acb4-4b03-4b46-92b4-f8869ba8241e,javaHome=/Library/Java/JavaVirtualMachines/jdk-11.0.11.jdk/Contents/Home,daemonRegistryDir=/Users/nabil/.gradle/daemon,pid=47119,idleTimeout=10800000,priority=NORMAL,daemonOpts=--add-opens,java.base/java.util=ALL-UNNAMED,--add-opens,java.base/java.lang=ALL-UNNAMED,--add-opens,java.base/java.lang.invoke=ALL-UNNAMED,--add-opens,java.base/java.util=ALL-UNNAMED,--add-opens,java.prefs/java.util.prefs=ALL-UNNAMED,--add-opens,java.prefs/java.util.prefs=ALL-UNNAMED,--add-opens,java.base/java.nio.charset=ALL-UNNAMED,--add-opens,java.base/java.net=ALL-UNNAMED,--add-opens,java.base/java.util.concurrent.atomic=ALL-UNNAMED,-Xmx2048M,-Dfile.encoding=UTF-8,-Duser.country=GB,-Duser.language=en,-Duser.variant]
ComposeOptions.kotlinCompilerVersion is deprecated. Compose now uses the kotlin compiler defined in your buildscript.
Kotlin Multiplatform Projects are an Alpha feature. See: https://kotlinlang.org/docs/reference/evolution/components-stability.html. To hide this message, add 'kotlin.mpp.stability.nowarn=true' to the Gradle properties.

Warning: The 'kotlin-android-extensions' Gradle plugin is deprecated. Please use this migration guide (https://goo.gle/kotlin-android-extensions-deprecation) to start working with View Binding (https://developer.android.com/topic/libraries/view-binding) and the 'kotlin-parcelize' plugin.

The Kotlin source set androidAndroidTestRelease was configured but not added to any Kotlin compilation. You can add a source set to a target's compilation by connecting it with the compilation's default source set using 'dependsOn'.
See https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#connecting-source-sets
Unable to strip the following libraries, packaging them as they are: libc++_shared.so, librealm-ffi-dbg.so, librealmc.so.
w: skipping /Users/nabil/.gradle/caches/modules-2/files-2.1/io.realm.kotlin/library-iosx64/0.45.0-SNAPSHOT/9f6174f4f610fd23472aa2f0e2d9879584c9ecb5/library.klib. Incompatible abi version. The current default is '1.4.2', found '1.5.0'. The library produced by 1.5.20 compiler
e: Could not find "/Users/nabil/.gradle/caches/modules-2/files-2.1/io.realm.kotlin/library-iosx64/0.45.0-SNAPSHOT/9f6174f4f610fd23472aa2f0e2d9879584c9ecb5/library.klib" in [/Users/nabil/Dev/realm/KMP/master-2/realm-kotlin-samples/Bookshelf, /Users/nabil/.konan/klib, /Users/nabil/.konan/kotlin-native-prebuilt-macos-1.5.10/klib/common, /Users/nabil/.konan/kotlin-native-prebuilt-macos-1.5.10/klib/platform/ios_x64]
Daemon vm is shutting down... The daemon has exited normally or was terminated in response to a user interrupt.
----- End of the daemon log -----


FAILURE: Build failed with an exception.

```

### CODE REVIEW

File: buildSrc/src/main/kotlin/Config.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ object Versions {
     const val gradlePluginPublishPlugin = "0.15.0" // https://plugins.gradle.org/plugin/com.gradle.plugin-publish
     const val junit = "4.13.2" // https://mvnrepository.com/artifact/junit/junit
     const val jvmTarget = "1.8"
-    const val kotlin = "1.5.20" // https://github.com/JetBrains/kotlin
+    const val kotlin = "1.5.10" // https://github.com/JetBrains/kotlin
```
Comment (cmelchior) [inline buildSrc/src/main/kotlin/Config.kt:74]: This doesn't affect other compiler plugins like atomic-fu?

**Review by cmelchior — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,11 +38,11 @@ This new architecture is intended to make it easier to consume and work with Rea
 * None.
 
 ### Compatibility
-* This release is compatible with Kotlin 1.5.20 and Coroutines 1.5.0.
+* This release is compatible with Kotlin 1.5.10 and Coroutines 1.5.0.
 
 ### Internal
 * Updated `com.gradle.plugin-publish` to 0.15.0.
-* Updated Kotlin support to 1.5.20.
+* Reverted to Kotlin `1.5.10` to fix a compatibility issue.
```
Comment (rorbech) [inline CHANGELOG.md:45]: Since the previous versions didn't use 1.5.20 I guess we didn't revert. Only reverted for the snapshots.

**Review by rorbech — APPROVED** [review]
Minor comment to the CHANGELOG.

File: CHANGELOG.md (hunk 2 above)
Comment (nhachicha) [reply CHANGELOG.md]: I'll remove the entry

**Pull request merged**